client/components/tinymce/plugins/media/plugin.jsx

```diff
@@ -226,6 +226,12 @@ function mediaButton( editor ) {
 			if ( current.media.transient ) {
 				transients++;
 				isTransientDetected = true;
+
+				// Mark the image as a transient in upload
+				editor.dom.$( img ).toggleClass( 'is-transient', true );
```

Comment:
It's because we don't want to hold this fix until we figure out a workaround ;) 

Nag me if nothing happens. ;)